### PR TITLE
feat: add support for talend-scripts.js as config

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
We have a case where we want to have variable Sass variables (injecting a CDN URL via environment for example).
A JSON file can't read env variables, so that's a limitation due to the file format.

**What is the chosen solution to this problem?**
Make `ui-scripts` support `talend-scripts.js` as configuration input.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/core/doc/presets.md [hunk 1]
```diff
@@ -47,15 +47,15 @@ const {
...
+2. Create a `talend-scripts` [configuration file](./customisation.md) at your project root.
 
 ```
 {
   "preset": ${presetName}
 }
 ```
 
-3. Complete `talend-scripts.json`, following your preset documentation.
+3. Complete `talend-scripts` configurationf ile, following your preset documentation.
```
Comment (ncomont) [inline packages/core/doc/presets.md:58]:
```suggestion
3. Complete `talend-scripts` configuration file, following your preset documentation.
```

File: packages/config-react-webpack/README.md [hunk 2]
```diff
@@ -13,6 +13,7 @@ Your folder hierarchy should follow
     |_ cmf.json
     |_ package.json
     |_ talend-scripts.json
+    |_ (talend-scripts.js)
```
Comment (tlnd-mhuchet) [inline packages/config-react-webpack/README.md:16]: It is one or the other, so maybe something like below to replace the two lines?
```suggestion
    |_ talend-scripts.js(on)
```

File: packages/config-react-webpack/README.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ Your folder hierarchy should follow
     |_ cmf.json
     |_ package.json
     |_ talend-scripts.json
```
Comment (frassinier) [inline packages/config-react-webpack/README.md:15]:
```suggestion

```

**Review by frassinier — APPROVED** [review]

File: packages/core/utils/env.js [hunk 4]
```diff
@@ -3,19 +3,27 @@ const get = require('lodash.get');
...
-	if (fs.existsSync(userConfigFilePath)) {
-		return require(userConfigFilePath);
+	const configFilePath = [
+		path.join(process.cwd(), 'talend-scripts.js'),
+		path.join(process.cwd(), 'talend-scripts.json'),
+	].find(file => fs.existsSync(file));
+
+	if (!configFilePath) {
+		// Load and return first config file found
+		return require(configFilePath);
```
Comment (jmfrancois) [inline packages/core/utils/env.js:17]: I guess there is a typo in the if condition

Comment (sgangnant) [reply packages/core/utils/env.js:17]: Oh yes thanks 🙃

**Review by ncomont — APPROVED** [review]

**Review by tlnd-mhuchet — APPROVED** [review]

**Review by gjou-tlnd — APPROVED** [review]

**Pull request merged**